### ISSUE TITLE
Logging

### ISSUE THREAD

**Issue by rwoll** [rwoll/submission#66]
If an error or other logging even gets called before the logger is attached, the *catch-all* error handlers crash the app since `req.log.error` since `req.log` is `undefined` (as encountered in #65 which lacks logging).

@mgree Should we create an "annonymous" logger that can be attached through `app.use(...)` at the very beginning of the app and used by a `catch-all` error handler that is registered before the `req.user`-aware logger?

**Comment by mgree** [comment]
That’s a good idea. Or just test for undefined, even—we can adapt my code:
```
        if (!req && env === 'development') {
          console.log(err);
        }
```
to something like:
```
        if ((!req || !req.log) && env === 'development') {
          console.log(err);
        } else {
          // the rest of the method
```

To avoid silliness.

On Jun 1, 2016, at 9:22 PM, Ross Wollman <notifications@github.com<mailto:notifications@github.com>> wrote:


If an error or other logging even gets called before the logger is attached, the catch-all error handlers crash the app since req.log.error since req.log is undefined (as encountered in #65<https://github.com/rwoll/submission/pull/65> which lacks logging).

@mgree<https://github.com/mgree> Should we create an "annonymous" logger that can be attached through app.use(...) at the very beginning of the app and used by a catch-all error handler that is registered before the req.user-aware logger?

**Comment by rwoll** [comment]
I don't mind the extra check for a fail-safe in dev-mode, but the `console.log(...)` won't go to bunyan which I think is important for figuring out where we run in to problems once deployed.

**Comment by rwoll** [comment]
Although, perhaps there is nothing worth permanently logging until a user is authenticated?

**Comment by mgree** [comment]
In production, if we’re that far off, we’re just done.

Yeah, that’s got a development test anyway. bunyan prints out logs it can’t parse as is, I believe.

On Jun 1, 2016, at 10:57 PM, Ross Wollman <notifications@github.com<mailto:notifications@github.com>> wrote:


Although, perhaps there is nothing worth permanently logging until a user is authenticated?

**Comment by mgree** [comment]
Resolved in #70.
